store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java

```diff
@@ -238,20 +249,17 @@ protected void resendCode(Account account, Mailbox mbox, ZimbraSoapContext zsc,
             if (expiryTime < now.getTime()) {
                 // generate new code and send
                 sendCode(email, resendCount + 1, account, mbox, zsc, octxt);
-
             } else {
                 // send existing code
-                long expiry = account.getRecoveryEmailCodeValidity();
-                long newExpiryTime = now.getTime() + expiry;
                 Account authAccount = getAuthenticatedAccount(zsc);
                 // update resend count
                 resendCount = resendCount + 1;
                 HashMap<String, Object> prefs = new HashMap<String, Object>();
-                prefs.put(Provisioning.A_zimbraRecoveryEmailVerificationData,
-                    email + ":" + code + ":" + newExpiryTime + ":" + resendCount);
-                Provisioning.getInstance().modifyAttrs(mbox.getAccount(), prefs, true,
-                    zsc.getAuthToken());
-                sendRecoveryEmailVerificationCode(authAccount, account, email, code, newExpiryTime,
+                dataMap.put(RESEND_COUNT, String.valueOf(resendCount));
+                String verificationDataStr = JWEUtil.getJWE(dataMap);
+                prefs.put(Provisioning.A_zimbraRecoveryEmailVerificationData, verificationDataStr);
```

Comment:
Could replace `verificationDataStr` with `JWEUtil.getJWE(dataMap)` and delete previous line.